src/internal/mod.rs

```diff
@@ -4,8 +4,9 @@
 
 //! Non exposed modules.
 
-pub mod assignment;
-pub mod core;
-pub mod incompatibility;
-pub mod memory;
-pub mod partial_solution;
+mod assignment;
```

Comment:
no need to change anything here, those are effectively already `pub(crate)` since `internal` module is `mod` and not `pub mod`

Reply:
If we are to make code reorganization at some point, let leave it for another PR.